test/transforms/test_pad.py

```diff
@@ -0,0 +1,536 @@
+
+            # Compare non-padded area with the original.
+            assert torch.equal(original_tensor,
+                               padded_tensor[:compare_pad_start_idx, :])
+
+
+def _check_data(original: Union[Data, HeteroData], padded: Union[Data,
+                                                                 HeteroData],
+                max_num_nodes: Union[int, Dict[NodeStoreType, int]],
+                max_num_edges: Optional[Union[int, Dict[EdgeStoreType,
+                                                        int]]] = None,
+                node_pad_value: Optional[Union[Padding, int, float]] = None,
+                edge_pad_value: Optional[Union[Padding, int, float]] = None,
+                exclude_keys: Optional[List[str]] = None) -> None:
+
+    if not isinstance(node_pad_value, Padding) and node_pad_value is not None:
+        node_pad_value = UniformPadding(node_pad_value)
+    if not isinstance(edge_pad_value, Padding) and edge_pad_value is not None:
+        edge_pad_value = UniformPadding(edge_pad_value)
+
+    if exclude_keys is None:
+        exclude_keys = []
+
+    if isinstance(original, Data):
+        _check_homo_data_nodes(original, padded, max_num_nodes, node_pad_value,
+                               exclude_keys)
+        _check_homo_data_edges(original, padded, max_num_edges, edge_pad_value,
+                               exclude_keys)
+    else:
+        _check_hetero_data_nodes(original, padded, max_num_nodes,
+                                 node_pad_value, exclude_keys)
+        _check_hetero_data_edges(original, padded, max_num_edges,
+                                 edge_pad_value, exclude_keys)
+
+
+def test_pad_repr():
+    pad_str = 'Pad(max_num_nodes=10, max_num_edges=15, ' \
+        'node_pad_value=UniformPadding(padding_values=3.0), ' \
+        'edge_pad_value=UniformPadding(padding_values=1.5))'
+    assert eval(pad_str).__repr__() == pad_str
+
+
+@pytest.mark.parametrize('data', [fake_data(), fake_hetero_data()])
+@pytest.mark.parametrize('num_nodes', [32, 64])
+def test_pad_auto_edges(data, num_nodes):
+    original = data
+    data = deepcopy(data)
+    transform = Pad(max_num_nodes=num_nodes)
+
+    padded = transform(data)
+    _check_data(original, padded, num_nodes)
+
+
+@pytest.mark.parametrize('num_nodes', [32, 64])
+@pytest.mark.parametrize('num_edges', [300, 411])
+def test_pad_data_explicit_edges(num_nodes, num_edges):
+    data = fake_data()
+    original = deepcopy(data)
+    transform = Pad(max_num_nodes=num_nodes, max_num_edges=num_edges)
+
+    padded = transform(data)
+    _check_data(original, padded, num_nodes, num_edges)
+
+
+@pytest.mark.parametrize('num_nodes', [32, {'v0': 64, 'v1': 36}])
+@pytest.mark.parametrize('num_edges', [300, {('v0', 'e0', 'v1'): 397}])
+def test_pad_heterodata_explicit_edges(num_nodes, num_edges):
+    data = fake_hetero_data()
+    original = deepcopy(data)
+    transform = Pad(max_num_nodes=num_nodes, max_num_edges=num_edges)
+
```

Comment:
from this test and the test below `test_pad_data_pad_values`, we pass `max_num_nodes` to the `Pad` transform. Does this mean that all types of nodes in the heterogeneous graph are padded to the same size?
For example, if we have `author, affiliated_with, institution` relation. In one batch, we have 1750 author nodes, 54 institution nodes. If it possible to pad author node to be 1800 for all batches and institution nodes to 60 for all batches? The same might apply on the edges. In this case, we don't waste space on institution nodes by padding it to 1800. Let me know if this makes sense or if I  understand this wrong.

Reply:
It makes sense and it is possible in the current implementation. See that in one test case of `test_pad_heterodata_explicit_edges`, `max_num_nodes` is `32` but in the other we pass a dictionary `{'v0': 64, 'v1': 36}` meaning that `v0` node is going to be padded to be `64` and `v1` -- to `36`. See also the description of the `max_num_nodes` argument in the `Pad` class:

> The number of nodes after padding.
In heterogeneous graphs, may also take in a dictionary denoting the
number of nodes for specific node types. The dictionary must
specify values for all the possible node types.

So in your case you could pass sth like `{'author': 1800, 'institution': 60}`. The same applies to edges, `max_num_edges` can also be a dictionary with values for specific edge types.

Reply:
That's excellent! Sounds good to me. Thanks for the explanation.